datafusion/scheduler/Cargo.toml

```diff
@@ -0,0 +1,57 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "datafusion-scheduler"
+description = "Scheduling for DataFusion query engine"
+version = "7.0.0"
+homepage = "https://github.com/apache/arrow-datafusion"
+repository = "https://github.com/apache/arrow-datafusion"
+readme = "../README.md"
+authors = ["Apache Arrow <dev@arrow.apache.org>"]
+license = "Apache-2.0"
+keywords = ["arrow", "query", "sql"]
+edition = "2021"
+rust-version = "1.58"
+
+[lib]
+name = "datafusion_scheduler"
+path = "src/lib.rs"
+
+[features]
+
+[dependencies]
+ahash = { version = "0.7", default-features = false }
+arrow = { version = "12" }
+async-trait = "0.1"
+datafusion = { path = "../core", version = "7.0.0" }
```

Comment:
Sounds good - this will also help down the line if we want to stabilise things :+1:

Reply:
So, I'm not sure how to do this without introducing a circular dependency. Currently ExecutionPlan is defined in datafusion/core. Options I can see are:

* Move ExecutionPlan into datafusion-common, this would require also moving the metrics
* Merge datafusion-scheduler with datafusion/core

I'm not sure which would be preferable?

Reply:
> Merge datafusion-scheduler with datafusion/core

This is kind of what I was expecting (and then conditionally compile it based on a feature flag)

Reply:
I would rather see the scheduler added in core than see physical plans (ExecutionPlan) move to datafusion-common